src/test/memory/segments_using_allocators_test.cpp

```diff
@@ -0,0 +1,111 @@
+#include "../storage/encoding_test.hpp"
+
+#include "base_test.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_encoded_segment.hpp"
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/value_segment.hpp"
+
+namespace opossum {
+
+// In these tests, we make sure that segment encoders properly use the assigned memory resource
+
+// A simple polymorphic memory resource that tracks how much memory was allocated
+class SimpleTrackingMemoryResource : public boost::container::pmr::memory_resource {
+ public:
+  size_t allocated{0};
+
+  void* do_allocate(std::size_t bytes, std::size_t alignment) override {
+    allocated += bytes;
+    return std::malloc(bytes);  // NOLINT
+  }
+
+  void do_deallocate(void* p, std::size_t bytes, std::size_t alignment) override {
+    allocated -= bytes;
+    std::free(p);  // NOLINT
+  }
+
+  bool do_is_equal(const memory_resource& other) const noexcept override { return false; }
```

Comment:
What is that about?

Reply:
Please elaborate. That's just the interface.

Reply:
`Fail()`